src/main/java/org/uast/astgen/codegen/java/NodeTypeConstructor.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * MIT License Copyright (c) 2022 unified-ast
+ * https://github.com/unified-ast/ast-generator/blob/master/LICENSE.txt
+ */
+package org.uast.astgen.codegen.java;
+
+import java.util.List;
+import org.uast.astgen.rules.Child;
+import org.uast.astgen.rules.Descriptor;
+import org.uast.astgen.rules.DescriptorAttribute;
+import org.uast.astgen.rules.Node;
+
+/**
+ * Generates type subclass source code for rules that describe nodes.
+ *
+ * @since 1.0
+ */
+final class NodeTypeConstructor extends NodeConstructor {
+    /**
+     * The 'List&lt;String&gt;' type.
+     */
+    private static final String LIST_STRING = "List<String>";
+
+    /**
+     * The 'List&lt;ChildDescriptor&gt;' type.
+     */
+    private static final String LIST_CHILD = "List<ChildDescriptor>";
+
+    /**
+     * The start of unmodifiable list declaration.
+     */
+    private static final String LIST_BEGIN = "Collections.unmodifiableList(Arrays.asList(";
+
+    /**
+     * The end of unmodifiable list declaration.
+     */
+    private static final String LIST_END = "))";
+
+    /**
+     * The list separator.
+     */
+    private static final String SEPARATOR = ", ";
+
+    /**
+     * Static string generator to avoid Qulice error messages.
+     */
+    private final StaticStringGenerator ssg;
+
+    /**
+     * Constructor.
+     * @param env The environment
+     * @param rule The rule
+     * @param klass The class to be filled
+     */
+    NodeTypeConstructor(final Environment env, final Node rule, final Klass klass) {
+        super(env, rule, klass);
+        this.ssg = new StaticStringGenerator(klass);
+    }
+
+    @Override
+    public void construct() {
+        final Node rule = this.getRule();
+        final Klass klass = this.getKlass();
+        final Method name = new Method("getName");
+        name.setReturnType("String");
+        name.setCode(String.format("return %s;", this.ssg.getFieldName(rule.getType())));
+        klass.addMethod(name);
+        this.fillChildTypes();
+        this.fillHierarchy();
+        final Method builder = new Method("createBuilder");
+        builder.setReturnType("Builder");
+        builder.setCode("return new Constructor();");
+        klass.addMethod(builder);
+    }
+
+    /**
+     * Fills in everything related to the child types list.
+     */
+    private void fillChildTypes() {
+        final Klass klass = this.getKlass();
+        this.createChildrenField();
+        final Method getter = new Method("getChildTypes");
+        getter.setReturnType(NodeTypeConstructor.LIST_CHILD);
+        getter.setCode("return TypeImpl.CHILDREN;");
+        klass.addMethod(getter);
+    }
+
+    /**
+     * Creates 'List&lt;ChildDescriptor&gt;' CHILDREN structure.
```

Comment:
You may use `{@code List<ChildDescriptor>}` to avoid html restrictions.

Reply:
Fixed